ethereum/rpc/websockets.go

```diff
@@ -411,15 +411,58 @@ func (api *pubSubAPI) subscribeNewHeads(wsConn *wsConn, subID rpc.ID) (pubsub.Un
 					continue
 				}
 
+				baseFee := types.BaseFeeFromEvents(data.ResultBeginBlock.Events)
+
+				cosmosHash := data.Header.Hash()
+
 				header := types.EthHeaderFromTendermint(data.Header, ethtypes.Bloom{}, baseFee)
+				if data.NumTxs == 0 {
+					header.TxHash = ethtypes.EmptyTxsHash
+				}
+
+				bloom := ethtypes.Bloom{}
+				for _, et := range data.ResultEndBlock.Events {
+					if et.Type != evmtypes.EventTypeBlockBloom {
+						continue
+					}
+
+					for _, attr := range et.Attributes {
+						if attr.Key == evmtypes.AttributeKeyEthereumBloom {
+							bloom = ethtypes.BytesToBloom([]byte(attr.Value))
+						}
+					}
+				}
+
+				result := map[string]interface{}{
+					"parentHash":       header.ParentHash,
+					"sha3Uncles":       header.UncleHash,
+					"miner":            header.Coinbase,
+					"stateRoot":        header.Root,
+					"transactionsRoot": "0x01",
```

Comment:
An accurate value is required here. If it can be ignored temporarily, it is recommended to give an empty hash.